### ISSUE TITLE
The role of optimizer in trades_loss

### ISSUE THREAD

**Issue by s110m** [yaodongyu/TRADES#29]
Hi, 
In trades_loss you have used the argument 'optimizer' and in line 77 you call 'optimizer.zero_grad()'. 
Was there a need for this? In which part of the calculation of trade_loss gradients of the model are updated that we need to zero them?
Thanks a lot.

**Comment by kenekoba** [comment]
you can check by yourself in the file `trades.py`, 

it's only used here:
https://github.com/yaodongyu/TRADES/blob/6e8e11b7c281371c2f027ffadfbaea80361f09de/trades.py#L77
